nixos/modules/services/misc/sonarr.nix

```diff
@@ -27,6 +39,79 @@ in
         '';
       };
 
+      apiKeyFile = lib.mkOption {
+        type = lib.types.nullOr lib.types.path;
+        description = ''
+          Path to the file containing the API key for Sonarr (32 chars).
+          This will overwrite the API key in the config file.
+        '';
+        example = "/run/secrets/sonarr-apikey";
+        default = null;
+      };
+
+      environmentFiles = lib.mkOption {
+        type = lib.types.listOf lib.types.path;
+        default = [ ];
+        description = ''
+          Environment file to pass secret configuration values.
+
+          Each line must follow the `SONARR__SECTION__KEY=value` pattern.
+          Please consult the documentation at the [wiki](https://wiki.servarr.com/useful-tools#using-environment-variables-for-config).
+        '';
+      };
+
+      settings = lib.mkOption {
+        type = lib.types.submodule {
+          freeformType = (pkgs.formats.ini { }).type;
+          options = {
+            update = {
+              mechanism = lib.mkOption {
+                type = with lib.types; nullOr str;
+                description = "which update mechanism to use";
+                default = "external";
```

Comment:
Was making the update.mechanism type an enum considered?

Reply:
so base on the nix-code it makes sense. 

source: https://github.com/Sonarr/Sonarr/blob/v5-develop/frontend/src/typings/Settings/General.ts#L1

But for the usage i guess it would never make sense to define `apt` or `docker` here? Therefore i will not define them.

its implemented. thanks